src/Common/src/TypeSystem/Interop/IL/NativeStructType.cs

```diff
@@ -127,7 +127,23 @@ public override TypeSystemContext Context
 
         private NativeStructField[] _fields;
         private InteropStateManager _interopStateManager;
+        private bool _hasInvalidLayout;
 
```

Comment:
Ah. Could you move `INonEmittableType` to `Common\src\TypeSystem\CodeGen` then and make `NativeStructType` implement it in `NativeStructType.CodeGen.cs`? I can't come up with a better scheme right now :(. It's desirable to be able to express that a certain type should never be loaded at runtime, but at the same time it's none of the core/shared typesystem's concerns.

Cc @davidwrighton